File: typescript/src/structural/facade/publisher/Validator.ts
import { Article } from "../interfaces/IArticle";


export function validator(article: Article): boolean{

    const requireKeys: string[] = ["title", "author", "content"]
    const forbiddenWords: string[] = ["sucks", "idiot", "stupid"]

    for (const key in requireKeys){
        if (!(key in article)){
            console.log(`The require key us missing: ${key}`)
            return false
        }
    }

    const content: string = article.content || "";
    if (forbiddenWords.some(word => content.includes(word))){
        console.log("Forbidden words found in content")
        return false
    }

    const title : string = article.title || "";
    if (content.length <= 200 && title.length <= 10){
        console.log("The amount of content and title is correct.")
        return false
    }

    console.log("Validation passed: Article is valid.")
    return true

}

File: c#/Structural/Facade/Publisher/Validator.cs
using Interface.IArticle;

namespace Publisher.Validator
{
    public static class Validator
    {
        public static bool Validate(IArticle article)
        {
            string[] forbiddenWords = { "sucks", "idiot", "stupid" };
            string[] requireKeys = { "title", "author", "content" };
            string content = article.Content ?? "";
            string title = article.Title ?? "";

            foreach (var key in requireKeys)
            {
                if (string.IsNullOrEmpty(key))
                {
                    Console.WriteLine($"The require key is missing: {key}");
                    return false;
                }
            }

            if (forbiddenWords.Any(word => content.Contains(word)))
            {
                Console.WriteLine("Forbidden words found in content");
            }

            if (content.Length <= 200 && title.Length <= 10)
            {
                Console.WriteLine("The amount of content and title is correct");
                return false;
            }

            Console.WriteLine("Validation passed: Article is valid");
            return true;
        }
    }
}